extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/intrumentation/vertx/RedisClientInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,174 @@
+package io.quarkus.opentelemetry.runtime.tracing.intrumentation.vertx;
+
+import static io.quarkus.opentelemetry.runtime.config.build.OTelBuildConfig.INSTRUMENTATION_NAME;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanKindExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.db.DbClientAttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.db.DbClientAttributesGetter;
+import io.opentelemetry.instrumentation.api.instrumenter.db.DbClientSpanNameExtractor;
+import io.opentelemetry.instrumentation.api.internal.AttributesExtractorUtil;
+import io.opentelemetry.semconv.SemanticAttributes;
+import io.vertx.core.Context;
+import io.vertx.core.spi.tracing.SpanKind;
+import io.vertx.core.spi.tracing.TagExtractor;
+import io.vertx.core.tracing.TracingPolicy;
+
+public class RedisClientInstrumenterVertxTracer implements
+        InstrumenterVertxTracer<RedisClientInstrumenterVertxTracer.CommandTrace, Object> {
+    private final Instrumenter<CommandTrace, Object> redisClientInstrumenter;
+
+    public RedisClientInstrumenterVertxTracer(final OpenTelemetry openTelemetry) {
+        InstrumenterBuilder<CommandTrace, Object> clientInstrumenterBuilder = Instrumenter.builder(
+                openTelemetry,
+                INSTRUMENTATION_NAME,
+                DbClientSpanNameExtractor.create(RedisClientAttributesGetter.INSTANCE));
```

Comment:
About the name... I'd like to be able to run the rests in order to see what else can we do...

Reply:
Now I am following everything as the OTEL auto instrumentation does for Jedis, Lettuce, Redisson and Vert.x Redis Client... the only thing missing is the `ServerAttributesGetter` and `NetworkAttributesGetter`... which would be good to have some info visible from vert.x redis client here... as the command reporter and some attributes are not visible... I would like to be able to use RedisURI to properly do the server attributes and also have access to NetSocket to be able to do the network attributes...

Reply:
I'm the de facto maintainer of the Vert.x Redis client these days :-) I would be hesitant about exposing the `RedisURI`, but I'd gladly expose `RedisURI.toString()`. I'm certainly not going to expose the `NetSocket`, but I'd gladly expose whatever information you need. I assume it would all go through the Vert.x tracing mechanism, so let me know what would you need in `io.vertx.redis.client.impl.CommandReporter.Tags` and I can make that happen.

Reply:
Hey @Ladicek having the `netSocket.remoteAddress().hostAddress()` and `netSocket.remoteAddress().port` exposed as different attributes, and `redisUri.socketAddress().host()` and `redisUri.socketAddress().port()` as different attributes as well would be the things that we need to have all properties to server and network attributes. We could also built the statement if we had the command arguments exposed, so the now we have the command name that is the operation, but the statement would be operation + arguments, and we would sanitize the arguments before creating the statement. I think that we can go as is right now, and later we can add those extra properties that will enhance the tracing

Reply:
One other thing would it be possible for the command reporter to be public? So we are able to check it with instance of since now we are checking for a string in the attributes to handle or not the redis tracing. And it would be awesome to have the tags enum available so in our side we can do the mapping conversion to what OTEL is expecting without having to duplicate the constants on Quarkus side, especially if something changes during a bump, which is prone to errors...

Reply:
https://github.com/vert-x3/vertx-redis-client/issues/441 <- Created the issue here